File: commons/src/main/java/com/eirsteir/coffeewithme/commons/security/JwtUtils.java
package com.eirsteir.coffeewithme.commons.security;

import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import java.util.Date;
import java.util.stream.Collectors;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;

public class JwtUtils {

  public static String createJwtToken(
      JwtConfig jwtConfig, Authentication auth, UserDetailsImpl principal) {
    long now = System.currentTimeMillis();
    return Jwts.builder()
        .setSubject(principal.getId().toString())
        .claim("user_id", principal.getId().toString())
        .claim("email", principal.getEmail()) // TODO: 02.06.2020 remove?
        .claim("nickname", principal.getNickname()) // TODO: 02.06.2020 remove?
        .claim(
            "authorities",
            auth.getAuthorities().stream()
                .map(GrantedAuthority::getAuthority)
                .collect(Collectors.toList()))
        .setIssuedAt(new Date(now))
        .setExpiration(new Date(now + jwtConfig.getExpiration() * 1000))
        .signWith(SignatureAlgorithm.HS256, jwtConfig.getSecret().getBytes())
        .compact();
  }
}

File: commons/src/main/kotlin/com/eirsteir/coffeewithme/commons/security/JwtUtils.kt
package com.eirsteir.coffeewithme.commons.security

import io.jsonwebtoken.Jwts
import io.jsonwebtoken.SignatureAlgorithm
import org.springframework.security.core.Authentication
import org.springframework.security.core.GrantedAuthority
import java.util.*
import java.util.stream.Collectors

class JwtUtils {

    companion object {
        fun createJwtToken(
            jwtConfig: JwtConfig, auth: Authentication, principal: UserDetailsImpl
        ): String {
            val now = System.currentTimeMillis()
            return Jwts.builder()
                .setSubject(principal.id.toString())
                .claim("user_id", principal.id.toString())
                .claim("email", principal.email) // TODO: 02.06.2020 remove?
                .claim("nickname", principal.nickname) // TODO: 02.06.2020 remove?
                .claim(
                    "authorities",
                    auth.authorities.stream()
                        .map { obj: GrantedAuthority -> obj.authority }
                        .collect(Collectors.toList()))
                .setIssuedAt(Date(now))
                .setExpiration(Date(now + jwtConfig.expiration * 1000))
                .signWith(SignatureAlgorithm.HS256, jwtConfig.secret.toByteArray())
                .compact()
        }
    }
}